pycqed/instrument_drivers/physical_instruments/QuTech_VSM_Module.py

```diff
@@ -0,0 +1,267 @@
+                         **kwargs)
+        self.modules = [1, 2, 3, 4, 5, 6, 7, 8]
+        self.channels = [1, 2, 3, 4]
+
+        self.add_parameters()
+
+        # example of how the commands could look
+        self.add_function('reset', call_cmd='*RST')
+
+        # Set socket option NODELAY, this should be handled by the base class.
+        if self._socket:
+            self._socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+
+        self.connect_message()
+
+    def _recv(self):
+        """ Overwrites base IP recv command to ensuring read till EOM
+            Again, this should be handled properly in a base class.
+        """
+        return self._socket.makefile().readline().rstrip()
+
+    def connect_message(self, idn_param='IDN', begin_time=None):
+        """ Log a standard message on initial connection to an instrument.
+
+            Overwritten from Instrument base class because it contained a naked
+            print() statement. When using logging, this is undesirable because
+            one cannot easily redirect or disable print().
+
+            Args:
+                idn_param (str): name of parameter that returns ID dict.
+                    Default 'IDN'.
+                begin_time (number): time.time() when init started.
+                    Default is self._t0, set at start of Instrument.__init__.
+        """
+        # start with an empty dict, just in case an instrument doesn't
+        # heed our request to return all 4 fields.
+        idn = {'vendor': None, 'model': None,
+               'serial': None, 'firmware': None}
+        idn.update(self.get(idn_param))
+        t = time.time() - (begin_time or self._t0)
+
+        logging.info('Connected to: {vendor} {model} '
+                     '(serial:{serial}, firmware:{firmware}) '
+                     'in {t:.2f}s'.format(t=t, **idn))
+
+    ###
+    # Generic SCPI commands
+    ###
+
+    def get_identity(self):
+        return self.ask('*IDN?')
+
+    def get_operation_complete(self):
+        return self.ask('*OPC?')
+
+    def get_error(self):
+        """ Returns:    "No error" or <error message>
+        """
+        return self.ask('system:error?')
+
+    def get_system_error_count(self):
+        return self.ask_int('system:error:count?')
+
+    # Parameters
+
+    def add_power_supply_parameters(self):
+        for power_supply in ['front', 'rear', 'phantom']:
+            name = 'powersupply_{p}'.format(p=power_supply)
+            getter = 'POWERSUPPLY:{p}?'.format(p=power_supply.upper())
+            setter = 'POWERSUPPLY:{p}'.format(p=power_supply.upper()) + ' {}'
+            # Disable setting the front power supply
+            if power_supply == 'front':
+                setter = None
+            self.add_parameter(name, get_cmd=getter, set_cmd=setter,
+                               vals=vals.OnOff())
+
+    def add_temperature_parameters(self):
+        # Command class temperature
+        self.add_parameter('setpoint',
```

Comment:
Also a docstring would be nice (can be added by docstring="") , I don't know the difference between the setpoint and average temperature.